fastly/FastlyPlugin.php

```diff
@@ -55,18 +55,24 @@ public function purgeFastlyCache()
   {
     $fastlyKey = $this->getSettings()->fastlyApiKey;
     $serviceId = $this->getSettings()->fastlyServiceId;
-    
+
+    $client = new \Guzzle\Http\Client();
+
     $url = 'https://api.fastly.com/service/' . $serviceId . '/purge_all';
-    
-    $options = array(
-      'http' => array(
-          'header'  => "Fastly-Key: " . $fastlyKey . "\r\n" .
-                       "Accept: application/json\r\n",
-          'method'  => 'POST'
-      ),
-    );
-
-    $context  = stream_context_create($options);
-    $result = file_get_contents($url, false, $context);
+
+    $headers = [
+      'Fastly-Key' => $fastlyKey,
+      'Accept' => 'application/json',
+    ];
+
+    $request = new \Guzzle\Http\Message\Request('POST', $url, $headers);
+
+    $response = $client->send($request);
+
+    if ($response->getStatusCode() == 200) {
+      return $response;
+    }
+
+    return 'Something went wrong. Status Code: ' . $response->getStatusCode();
```

Comment:
sorry, i wasn't very clear

i'm worried that returning a string, which in php is a truthy value, could be interpreted by whatever's calling it as a success. for example, if this were called like:

```
if (craft()->plugins->getPlugin('fastly')->purgeFastlyCache()) {
  //yay! purged!
} else {
  //error
}
```

the error block would never be called.

_however_, since the function before this diff had no return value at all, this isn't a step in the wrong direction! i'd say we have two options:

1. return the `$response` on success, and either `false` or `null` on failure, and log the message using `FastlyPlugin::log(' ... ', LogLevel::Error)`
2. return the `$response` in all cases, regardless of the status code, allowing the caller of this function to determine how to handle errors

again, because the function before this diff was always returning null, it doesn't matter too much which we choose, though i think the latter is more descriptive